Actually install sdkManager if it is not available

ViliusSutkus89 opened the pull request:
GitHub Actions have Android SDK preinstalled, but there's no sdkmanager.
Current implementation of setup-android only checks if Android SDK is installed.
Missing sdkmanager means that no additional packages can be easily installed by downstream users.

Proposed solution would check if sdkmanager is available.
If sdkmanager is available - check if the version is correct (3.0). If the version is wrong (possibly too old) - download new version through the old available sdkmanager.
If sdkmanager is not available - download commandline-tools (containing sdkmanager) and move the whole package to where it should be, as if it was installed by sdkmanager (${android-sdk}/cmdline-tools/3.0/bin/sdkmanager).

After installing, add ${android-sdk}/cmdline-tools/3.0/bin/ to PATH, so that downstream users could just execute sdkmanager without needing to know it's path or --sdk_root argument.

This would allow downstream users to easily install extra packages, for example NDK (issue #67)
```yaml
- uses: android-actions/setup-android@version

- run: sdkmanager "ndk;21.0.6113669" "cmake;3.10.2.4988404" "emulator"
```
License acceptance, tool and platform-tools installing are done unconditionally.
If they are already installed, sdkmanager won't install them again.

Pro tips and remarks are much appreciated.

daveol commented on src/main.ts:
```diff
@@ -14,69 +15,140 @@ const COMMANDLINE_TOOLS_LIN_URL = `https://dl.google.com/android/repository/comm
 const ANDROID_HOME_DIR = path.join(HOME, '.android')
 const ANDROID_HOME_SDK_DIR = path.join(ANDROID_HOME_DIR, 'sdk')
-const ANDROID_REPOSITORIES_CFG = path.join(ANDROID_HOME_DIR, 'repositories.cfg')
+const ANDROID_SDK_ROOT = process.env['ANDROID_SDK_ROOT'] || ANDROID_HOME_SDK_DIR
```
II believe this is incompatible with current ubuntu builders, since the building process has no rights to write to the directory where the sdk is located

daveol reviewed (changes requested):
overall seems good, except for the location. Maybe we can symlink existing components to a place we can write to? and you dind't run a build using `npm run build`

ViliusSutkus89 commented:
I kinda recall something about not being able to write to SDK location too.
Added "runSdkManager" test to build-test.yml, which tests if install was sucessful and we can run sdkmanager now.
Turns out sdkmanager install is sucessful and we can write to SDK location just fine.

As I did the testing I found out that Windows2016 install is broken too, because SDK is installed to
`C:\Program Files (x86)\...`. That is bad, because lots of dev software don't like spaces in paths.
sdkmanager just wouldn't run. Added a check in `run(){ ... }` to move the SDK to a location without
spaces. Location is on the same drive, so the move is fast. `$env.ANDROID_SDK_ROOT` is set at the end, so it should be all good.

As for the symlink suggestion. File move command barely works on Windows runners, I'm afraid to even try symlinks there.
Kinda recall something about needing admin privileges to create symlinks Windows 10 too.

daveol commented:
I merged this manually
